sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryPartA.cs

```diff
@@ -155,6 +156,18 @@ internal static void InitRoleInfo(Resource resource)
             {
                 RoleName = serviceName;
             }
+
+            if (RoleInstance == null)
+            {
+                try
+                {
+                    RoleInstance = Dns.GetHostName();
```

Comment:
1) In case of NETFRAMEWORK, it appends domain name as well. Should we add that? (would be mapping change). Not sure why we do that for only NETFRAMEWORK though.
2) This will be a fall back - I think the primary way should be via resource.

Reply:
Will be merging this now. I think the domain part is not needed at this point.